scanner/src/main/kotlin/Scanner.kt

```diff
@@ -152,17 +152,34 @@ fun scanOrtResult(
 
     val endTime = Instant.now()
 
-    val filteredScannerOptions = packageScanner.scannerConfig.options?.let { options ->
-        options[packageScanner.scannerName]?.let { scannerOptions ->
-            val filteredScannerOptions = packageScanner.filterOptionsForResult(scannerOptions)
-            options.toMutableMap().apply { put(packageScanner.scannerName, filteredScannerOptions) }
+    // Note: Currently, each scanner gets its own reference to the whole scanner configuration, which includes the
+    // options for all scanners.
+    check(packageScanner.scannerConfig === projectScanner.scannerConfig)
+
+    val filteredScannerConfigOptions = mutableMapOf<String, ScannerOptions>()
```

Comment:
I actually had it like that, but then found it to sound too much like "filtered `ScannerOptions`" instead of a "filtered scanner configuration" (which is a map of `ScannerOptions`... but then again that's probably giving it too much thought. I guess I'll rename it again.